components/cam/src/physics/crm/sam/crm_output_module.F90

```diff
@@ -0,0 +1,581 @@
+      real(crm_rknd), allocatable :: md_crm(:,:)      ! mass flux down
+      real(crm_rknd), allocatable :: du_crm(:,:)      ! mass detrainment from updraft
+      real(crm_rknd), allocatable :: eu_crm(:,:)      ! mass entrainment from updraft
+      real(crm_rknd), allocatable :: ed_crm(:,:)      ! mass detrainment from downdraft
+      real(crm_rknd), allocatable :: jt_crm(:)        ! index of cloud (convection) top
+      real(crm_rknd), allocatable :: mx_crm(:)        ! index of cloud (convection) bottom
+
+      ! Other stuff...
+      real(crm_rknd), allocatable :: flux_qt      (:,:)  ! nonprecip water flux        [kg/m2/s]
+      real(crm_rknd), allocatable :: fluxsgs_qt   (:,:)  ! sgs non-precip water flux   [kg/m2/s]
+      real(crm_rknd), allocatable :: tkez         (:,:)  ! tke profile                 [kg/m/s2]
+      real(crm_rknd), allocatable :: tkesgsz      (:,:)  ! sgs tke profile             [kg/m/s2]
+      real(crm_rknd), allocatable :: tkz          (:,:)  ! tk profile                  [m2/s]
+      real(crm_rknd), allocatable :: flux_u       (:,:)  ! x-momentum flux             [m2/s2]
+      real(crm_rknd), allocatable :: flux_v       (:,:)  ! y-momentum flux             [m2/s2]
+      real(crm_rknd), allocatable :: flux_qp      (:,:)  ! precipitating water flux    [kg/m2/s or mm/s]
+      real(crm_rknd), allocatable :: precflux     (:,:)  ! precipitation flux          [m/s]
+      real(crm_rknd), allocatable :: qt_ls        (:,:)  ! tend of nonprec water due to large-scale   [kg/kg/s]
+      real(crm_rknd), allocatable :: qt_trans     (:,:)  ! tend of nonprec water due to transport     [kg/kg/s]
+      real(crm_rknd), allocatable :: qp_trans     (:,:)  ! tend of    prec water due to transport     [kg/kg/s]
+      real(crm_rknd), allocatable :: qp_fall      (:,:)  ! tend of    prec water due to fall-out      [kg/kg/s]
+      real(crm_rknd), allocatable :: qp_src       (:,:)  ! tend of    prec water due to conversion    [kg/kg/s]
+      real(crm_rknd), allocatable :: qp_evp       (:,:)  ! tend of    prec water due to evp           [kg/kg/s]
+      real(crm_rknd), allocatable :: t_ls         (:,:)  ! tend of lwse  due to large-scale           [kg/kg/s] ???
+      real(crm_rknd), allocatable :: prectend     (:)    ! column integrated tend in precip water+ice [kg/m2/s]
+      real(crm_rknd), allocatable :: precstend    (:)    ! column integrated tend in precip ice       [kg/m2/s]
+      real(crm_rknd), allocatable :: taux     (:)    ! zonal CRM surface stress perturbation      [N/m2]
+      real(crm_rknd), allocatable :: tauy     (:)    ! merid CRM surface stress perturbation      [N/m2]
+      real(crm_rknd), allocatable :: z0m          (:)    ! surface stress                             [N/m2]
+      real(crm_rknd), allocatable :: timing_factor(:)    ! crm cpu efficiency
+
+#ifdef MAML
+      ! MAML variables
+      real(crm_rknd), allocatable :: crm_pcp(ncrms,crm_nx,crm_ny) ! CRM precip rate for MAML (m/s)
```

Comment:
Didn't intend to review this module.  Created a test case anyway with compset F-MAM2-MAML, and picked up these coding mistake: crm_pcp and crm_snw, declaraed as allocatable, should give just shape, no actual sizes for the dimensions.